File: DataVault/DataVault/Helpers/SnackBarHelper.m
//
//  SnackBarHelper.m
//  DataVault
//
//  Created by Assassin on 6/6/20.
//  Copyright © 2020 Dan Gerchcovich. All rights reserved.
//

#import <Foundation/Foundation.h>

#import "NotificationHelpers.h"
#import "SnackBarHelper.h"
#import <MaterialComponents/MaterialSnackbar.h>

@implementation SnackBarHelper

+(void) showSnackBarWithMessage:(NSString *)message{
    MDCSnackbarMessage* messageItem = [[MDCSnackbarMessage alloc] init];
    messageItem.text = message;
    messageItem.buttonTextColor = UIColor.greenColor;
    messageItem.duration = 0.5f;
    
    MDCSnackbarMessageAction *action = [[MDCSnackbarMessageAction alloc] init];
    action.title = @"Dismiss";

    messageItem.action = action;
    [messageItem setShouldDismissOnOverlayTap:(true)];
    
    MDCSnackbarManager* snackBar = [[MDCSnackbarManager alloc] init];
    [snackBar showMessage:messageItem];
    
    [NotificationHelpers generateLocalNotification:message];
}

+(void) showSnackBarWithActionedMessage:(NSString *) message invokedAction:(void (^)())actionInvoked {
    MDCSnackbarMessage* messageItem = [[MDCSnackbarMessage alloc] init];
    messageItem.text = message;
    messageItem.buttonTextColor = UIColor.greenColor;
    messageItem.duration = 0.5f;
    
    MDCSnackbarMessageAction *action = [[MDCSnackbarMessageAction alloc] init];
    action.handler = actionInvoked;
    action.title = @"Done";
    
    messageItem.action = action;
    [messageItem setShouldDismissOnOverlayTap:(true)];
    
    MDCSnackbarManager* snackBar = [[MDCSnackbarManager alloc] init];
    [snackBar showMessage:messageItem];
    [NotificationHelpers generateLocalNotification:message];
}

+(void) showSnackBarWithCustomBtnActionedMessage:(NSString *) message buttonTitle:(NSString *)buttonTtl invokedAction:(void (^)())actionInvoked {
    MDCSnackbarMessage* messageItem = [[MDCSnackbarMessage alloc] init];
    messageItem.text = message;
    messageItem.buttonTextColor = UIColor.greenColor;
    messageItem.duration = 0.5f;
    
    MDCSnackbarMessageAction *action = [[MDCSnackbarMessageAction alloc] init];
    action.handler = actionInvoked;
    action.title = buttonTtl;
    
    messageItem.action = action;
    [messageItem setShouldDismissOnOverlayTap:(true)];
    
    MDCSnackbarManager* snackBar = [[MDCSnackbarManager alloc] init];
    [snackBar showMessage:messageItem];
    [NotificationHelpers generateLocalNotification:message];
}

@end

File: SWAPIConsumer.Swift/SWAPIConsumer.Swift/Helpers/SnackbarHelper.swift
//
//  SnackbarHelper.swift
//  SWAPIConsumer.Swift
//
//  Created by Assassin on 10/6/20.
//  Copyright © 2020 Dan Gerchcovich. All rights reserved.
//

import Foundation

import TopViewControllerDetection;
//Material
import MaterialComponents.MaterialSnackbar;

class SnackbarHelper{

    static func showSnackBarWithMessage(message: String){

        let snackMessage = MDCSnackbarMessage(text: message);
        snackMessage.text = message;
        snackMessage.buttonTextColor = UIColor.green;
        snackMessage.duration = 1;
        
        let messageAction = MDCSnackbarMessageAction();
        messageAction.title = "DISMISS";
        snackMessage.action = messageAction;
        
        let manager = MDCSnackbarManager()
        manager.setButtonTitleColor(UIColor.green, for: UIControl.State.normal);
        manager.setButtonTitleColor(UIColor.green, for: UIControl.State.highlighted);
        manager.setButtonTitleColor(UIColor.green, for: UIControl.State.selected);
        manager.show(snackMessage);
    }
    
    
    static func showSnackBarWithMessageWithCustomAction(message: String, actionHandler: @escaping () -> Void){

        let snackMessage = MDCSnackbarMessage(text: message);
        snackMessage.text = message;
        snackMessage.duration = 1;
        
        let messageAction = MDCSnackbarMessageAction();
        messageAction.title = "DISMISS";
        messageAction.handler =  (() -> Void)?{
            actionHandler();
        };
        
        snackMessage.action = messageAction;
        
        let manager = MDCSnackbarManager()
        manager.setButtonTitleColor(UIColor.green, for: UIControl.State.normal);
        manager.setButtonTitleColor(UIColor.green, for: UIControl.State.highlighted);
        manager.setButtonTitleColor(UIColor.green, for: UIControl.State.selected);
        manager.show(snackMessage);
    }
}
